eth/stagedsync/bor_heimdall_shared.go

```diff
@@ -220,19 +217,21 @@ func fetchAndWriteHeimdallStateSyncEvents(
 			continue
 		}
 
-		if lastStateSyncEventID+1 != eventRecord.ID || eventRecord.ChainID != chainID || !eventRecord.Time.Before(to) {
+		if lastStateSyncEventID+1 != eventRecord.ID || eventRecord.ChainID != chainID ||
+			!(eventRecord.Time.After(from) && eventRecord.Time.Before(to)) {
```

Comment:
I'd suggest to add a new unit test for this error path - it should be fairly easy to add.

You can have a look at `TestBorHeimdallForwardPersistsStateSyncEvents` and `stagedsynctest.Harness.mockHeimdallClient`. For this you will need to pass in a new function to `h.heimdallClient.EXPECT().FetchStateSyncEvents(...).DoAndReturn()`. A clean way to do this may be to add an optional `MockFetchStateSyncEventsCallback` to the `HarnessCfg` and check if it is present use that inside the DoAndReturn otherwise use the current one.

Reply:
ok I'll look at this tomorrow.